### ISSUE TITLE
Whatsapp <-> Email bridge

### ISSUE THREAD

**Issue by ripper234** [WA-WF-Bot/WA-WF-Bot-Public#42]
As a user who does not use Workflowy, but rather only use Email & Whatsapp:

I would like to write a Whatsapp to Fredo, and have Fredo email me this message.
(Also image upload & voice+transcription as usual).

**Comment by liatzionimorad** [comment]
will check

**Comment by liatzionimorad** [comment]
#24 is the same

**Comment by ripper234** [comment]
#24 is not the same - it's for incoming channel.

This issue is for **outgoing** channel (instead of WF).

**Comment by ripper234** [comment]
This would be our 2nd backend.

Right now, we have one frontend (Whatsapp), and one backend (Workflowy).

This feature would add a second backend - email.

In the future, we would want to support multiple concurrent backends for the same user.
But for now - **only one backend is supported per user**.

A new user sends the first message to the bot ("hi").

In the welcome reply we send it - we need the user to choose whether he wants to use Workflowy or Email
E.g.

"
Where do you manage your tasks?
1. Email
2. Workflowy
3. Somewhere else     (option 3 should ask the user "Where?", and report the data to us in some manner. Out of scope).

Assuming the user replies "1" or "email" (case insensitive), then the user selects the Email backend.

We ask the user "Please enter your email"
After the user gives us the their email, we send them a **verification email** from bot@fredo.ai with a unique link & code.
The bot then says
"
We sent you a verification link. What's the code?
(if you don't find it, check your spam folder)
"

The user can verify the code by either clicking on the verification link in the email, or by entering the code to the bot.
(Please test both cases to make sure the flow makes sense).

Once the user has verified their email, the bot writes something like "success" (like the usual success flow in Workflowy today).

From now, everything the user write to the bot (except special ! commands) is sent to the user's email.

**Email Sender**
Fredo display name (with a bot icon)
bot@fredo.ai

**Email Subject**
"Note - [First 16 characters of the message]"

**Email Body**
Text - Should contain the entire message from the bot.
Image - Should contain the image inline (not as attachment)
Audio - should contain an attachment with the original audio. The text body of the email should be a transcription. (When that feature is developed).

**Comment by elihaidv** [comment]
Usually we use this third party to send emails:
https://sendgrid.com/

I need an account to this service.

**Comment by liatzionimorad** [comment]
@sgino209 feedback:
The idea is clear. Admin’s Sendgrid account may indeed be applied for launching the corresponding emails. From a user's perspective, we can consider adding a staggering option, i.e. send the email only after N messages (set by user, with default=1) or when user explicitly triggers to send the email.

**Comment by liatzionimorad** [comment]
@elihaidv please check if we can insert the flexibility Shahar suggested.( the staggering option as a part of the command)

**Comment by ripper234** [comment]
@assafza can you review? We increased to priority-1.

**Comment by assafza** [comment]
This seems pretty straight forward and clear, but the question here for me is why Gmail and not Google keep? Or why Gmail and not Any.Do? or other platforms. Is there a good reason for choosing Gmail?
In any case I think we can find ways to validate those assumptions without or with very little coding involved.

See what I wrote in issue #57 as a suggestion to test it before developing it. LMK what you think.

**Comment by elihaidv** [comment]
For begging we can send one email per message.
In the future we can add option to aggregate messages in one mail. Or send in specific time. (If needed)

**Comment by liatzionimorad** [comment]
@assafza  can you relate to the flow please ( what should be asked from the user ). Please relate to  what Ron already described  add if needed.
To my  view at this point support sending to one mail address per user.

**Comment by assafza** [comment]
I have added a flow to Miro: https://miro.com/app/board/o9J_lttkfEA=/?invite_link_id=914151478716
Also, image attached below.
![Screen Shot 2022-02-16 at 20 59 57](https://user-images.githubusercontent.com/25544824/154336647-c8b8bd0c-c616-45a2-b05b-34b15a790bee.png)
